recipe-client-addon/test/browser/Utils.jsm

```diff
@@ -27,4 +27,48 @@ this.Utils = {
       yield testGenerator(driver);
     });
   },
+
+  /**
+   * Test wrapper that replaces a property on an object with another value for
+   * the duration of the test. The test generator is passed the replaced value.
+   */
+  withPatch(object, propertyName, patchValue, testGenerator) {
+    return function* inner() {
+      const original = object[propertyName];
+      object[propertyName] = patchValue;
+      yield testGenerator(patchValue);
+      object[propertyName] = original;
+    };
+  },
+
+  /**
+   * Creates a wrapper function that tracks whether it has been called.
+   */
+  createMock(wrapped) {
```

Comment:
YAY WE HAVE SINON IN TREE

Like I wanted to include it myself but was hemming and hawwing about 50kb of JS for like one bit of its functionality.